src/presentation/express-error-handler.ts

```diff
@@ -3,11 +3,18 @@ import { Exception } from '../model/exceptions/exception'
 import { ErrorCode } from '../model/enums/error-code'
 import { HttpStatusCode } from './http-status-code'
 import { HttpErrorHandler } from './interfaces/http-error-handler'
+import { Logger } from '../logger'
 
 export class ExpressErrorHandler implements HttpErrorHandler {
+
+  private readonly logger: Logger
+
+  constructor(logger: Logger) {
+    this.logger = logger.tag(ExpressErrorHandler.name)
+  }
+
   public handleError(response: Response, exception: Exception): void {
-    console.log(`Caught Exception: "${exception.errorCode}". Message: ${exception.message}`)
-    console.log(exception.stack)
+    this.logger.data(exception.stack).error(`Caught Exception: "${exception.errorCode}". Message: ${exception.message}`)
```

Comment:
Can't we just give the exception as data? It would be nice if the error message gives insight in what the context was when this happened in a human readable format. One way to achieve this would be to have the `handleError` method take an extra (required) argument that is the error message.

Reply:
I simply updated the log statement to remain as close to the original as possible.
Has now been updated to have the complete exception in the data.

Not sure about the idea of the extra required message argument.
The method is only ever called from a catch clause, where anything might have happend to end up there.
The message would not be able to add much more detail, not already included by the logged exception. I guess being told where the call comes from does have some merit, as the tag will just say `<ExpressErrorHandler>``as it is currently.

If an extra argument is to be added, then maybe it should specify who the caller is, to add that to the log. Can the logger have multiple Tags?